src/main/java/io/axoniq/axonserver/connector/query/impl/QueryChannelImpl.java

```diff
@@ -270,30 +292,26 @@ public Registration registerQueryHandler(QueryHandler handler, QueryDefinition..
                         result = CompletableFuture.allOf(result, sendUnsubscribe(def));
                         logger.debug("Unregistered handlers for query '{}' in context '{}'", def, context);
                     }
+                    supportedQueries.computeIfPresent(def, (qd, counter) -> counter.decrementAndGet() == 0 ? null : counter);
```

Comment:
If we decrement to 0, shouldn't we remove the entry from the `supportedQueries` entirely instead of removing the `AtomicInteger`?

Reply:
That' exactly what returning `null` from this method will do. See javadoc on `computeIfPresent`.

Reply:
Sweet, learning something new everyday. Sounds resolved too then.